apps/spruce/src/pages/configurePatch/index.tsx

```diff
@@ -34,30 +33,23 @@ const ConfigurePatch: React.FC = () => {
   usePageTitle(`Configure Patch`);
 
   // Can't configure a mainline version so should redirect to the version page
-  // @ts-expect-error: FIXME. This comment was added by an automated script.
-  if (!validateObjectId(patchId)) {
-    // @ts-expect-error: FIXME. This comment was added by an automated script.
-    return <Navigate to={getVersionRoute(patchId)} />;
+  if (!validateObjectId(patchId || "")) {
+    return <Navigate to={getVersionRoute(patchId || "")} />;
```

Comment:
I wonder if it's possible to swap the if-statements:
```
  if (loading) {
    return <PatchAndTaskFullPageLoad />;
  }

  if (error || !patch) {
    return <PageDoesNotExist />;
  }

  if (!validateObjectId(patch.id)) {
    return <Navigate to={getVersionRoute(patch.id)} />;
  }

  ...
```

Reply:
We don't need to rely on `patch.id` since that's the result of the query and it should be the same value with what ever is in `patchId` I did however find a way to optimize this page by skipping the `PATCH_CONFIGURE` query if `patchId` isn't a valid ObjectID`